sympy/physics/continuum_mechanics/column.py

```diff
@@ -0,0 +1,330 @@
+        # the bottom always zero
+        self._boundary_conditions['deflection'].append((0, 0))
+        self._boundary_conditions['slope'].append((0, 0))
+
+        if self._end_conditions['top'] == "pinned" and self._end_conditions['bottom'] == "pinned":
+            self._boundary_conditions['deflection'].append((self._height, 0))
+
+        elif self._end_conditions['top'] == "fixed" and self._end_conditions['bottom'] == "fixed":
+            # `M` is the reaction moment
+            M = Symbol('M')
+            self._boundary_conditions['deflection'].append((self._height, 0))
+            # moment  = P*y - M
+            self._moment -= M
+
+        elif self._end_conditions['top'] == "pinned" and self._end_conditions['bottom'] == "fixed":
+            # `F` is the horizontal force at the pinned end to counter the rection moment at fixed end
+            F = Symbol('F')
+            self._boundary_conditions['deflection'].append((self._height, 0))
+            # moment = P*y - F(l - x)
+            self._moment -= F*(self.height - x)
+
+        elif self._end_conditions['top'] == "free" and self._end_conditions['bottom'] == "fixed":
+            # `d` is the deflection at the free end
+            d = Symbol('d')
+            self._boundary_conditions['deflection'].append((self._height, d))
+            # moment = P*y - P*d
+            self._moment -= P*d
+
+        else:
+            raise ValueError("{} {} end-condition is not supported".format(sstr(self._end_conditions['top']), sstr(self._end_conditions['bottom'])))
+
+
+    def solve_slope_deflection(self):
+        """
+        Solves the differnetial equation of buckling to determine the
+        deflection and slope equations.
+
+        Examples
+        ========
+
+        A column of timber section is 8m long both ends being fixed.
+        Young's modulus of the timber is E and second-moment I.
+        The applied load on the column is 360KN.
+
+        >>> from sympy.physics.continuum_mechanics.column import Column
+        >>> from sympy import Symbol, symbols
+        >>> E, I, P = symbols('E, I, P', positive=True)
+        >>> c = Column(8, E, I, 360000, top="fixed", bottom="fixed")
+        >>> c.end_conditions
+        {'bottom': 'fixed', 'top': 'fixed'}
+        >>> c.boundary_conditions
+        {'deflection': [(0, 0), (8, 0)], 'slope': [(0, 0)]}
+        >>> c.moment()
+        -M + 360000*y(x)
+        >>> c.solve_slope_deflection()
+        >>> c.deflection()
+        -M*cos(600*x/(sqrt(E)*sqrt(I)))/360000 + M/360000
+        >>> c.slope()
+        M*sin(600*x/(sqrt(E)*sqrt(I)))/(600*sqrt(E)*sqrt(I))
+        """
+        y = Function('y')
+        x = self._variable
+        P = Symbol('P', positive=True)
+
+        C1, C2 = symbols('C1, C2')
+        E = self._elastic_modulus
+        I = self._second_moment
+
+        # differnetial equation of Column buckling
+        eq = E*I*y(x).diff(x, 2) + self._moment
+
+        defl_sol = dsolve(Eq(eq, 0), y(x)).args[1]
```

Comment:
You should check that the solution has been returned in explicit form (that args[0] is just the function on the lhs of the equation).

Reply:
Does this mean to check whether args[0] comes out to be `y(x)`?